src/component/Header/Header.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+interface HeaderProps {
+  todos: Todo[];
+  addTodo: (title: string) => void;
+  query: string;
+  setQuery: (query: string) => void;
+  areAllTasksCompleted: boolean;
+  handleAllTasksCompleted: (value: boolean) => void;
+}
+
+export const Header: React.FC<HeaderProps> = ({
+  todos,
+  addTodo,
+  query,
+  setQuery,
+  areAllTasksCompleted,
+  handleAllTasksCompleted,
+}) => {
+  const handleSubmit = (event: React.SyntheticEvent) => {
+    event.preventDefault();
+    addTodo(query);
+    setQuery('');
+  };
+
+  const handleToggleAll = () => {
+    handleAllTasksCompleted(!areAllTasksCompleted);
+  };
+
+  return (
+    <header className="todoapp__header">
```

Comment:
Consider using more descriptive class names like 'header__todo-app' instead of 'todoapp__header'.